### ISSUE TITLE
audio.pipe is not a function

### ISSUE THREAD

**Issue by lumalisan** [elevenlabs/elevenlabs-js#69]
I'm using literally the example from the official website (https://github.com/elevenlabs/elevenlabs-examples/blob/main/examples/text-to-speech/node/src/text_to_speech_file.ts) and I get this error: “audio.pipe is not a function”. Any idea? I'm using the same code, I haven't touched anything.

My package.json:
`{
  "version": "1.0.0",
  "description": "",
  "scripts": {
    "start": "ts-node src/index.ts"
  },
  "keywords": [],
  "dependencies": {
    "dotenv": "^16.4.5",
    "elevenlabs": "^0.11.0",
    "fluent-ffmpeg": "^2.1.3",
    "pexels": "^1.4.0",
    "uuid": "^10.0.0",
    "vosk": "^0.3.39"
  },
  "devDependencies": {
    "@types/node": "^20.14.12",
    "@types/uuid": "^10.0.0",
    "@types/dotenv": "^8.2.0",
    "@types/fluent-ffmpeg": "^2.1.24",
    "@types/vosk": "^0.3.1",
    "ts-node": "^10.9.2",
    "typescript": "^5.5.4"
  }
}`

**Comment by RohinBhargava** [comment]
We have pushed the code changes, and we need to run it through CI/Release processes; will keep you posted when it's ready!

**Comment by RohinBhargava** [comment]
Hey there, we have pushed a fix. It also appears the examples are out of date with the spec. Here is a ts example I am able to reproduce in node 16-21. Please let me know on this thread or directly if you have any trouble!

```
import { ElevenLabsClient } from "elevenlabs";
import { createWriteStream } from "fs";

import { v4 as uuid } from "uuid";

import * as dotenv from "dotenv";
import { chooseStreamWrapper } from "./src/core/fetcher/stream-wrappers/chooseStreamWrapper";

dotenv.config();

const ELEVENLABS_API_KEY = "9ea39b6927545e1c693bab163801f899";

const client = new ElevenLabsClient({
  apiKey: ELEVENLABS_API_KEY,
});

export const createAudioFileFromText = async (
  text: string
): Promise<string> => {
  return new Promise<string>(async (resolve, reject) => {
    try {
      const audio = await client.voiceGeneration.generate({
        gender: "female",
        accent: "american",
        age: "middle_aged",
        accent_strength: 1.0,
        text,
      });
      const fileName = `${uuid()}.mp3`;
      const fileStream = createWriteStream(fileName);

      (await chooseStreamWrapper(audio)).pipe(fileStream);
      fileStream.on("finish", () => resolve(fileName)); // Resolve with the fileName
      fileStream.on("error", reject);
    } catch (error) {
      reject(error);
    }
  });
};

createAudioFileFromText("Hello, World! Generate some random text that's more than 100 characters long and less than 1000 characters long.");
```

**Comment by RohinBhargava** [comment]
Hi there, we have pushed an update, and there is a workaround like this:
```
import { ElevenLabsClient } from "elevenlabs";
import { createWriteStream, writeFileSync } from "fs";

import { v4 as uuid } from "uuid";

import * as dotenv from "dotenv";

dotenv.config();

const ELEVENLABS_API_KEY = "9ea39b6927545e1c693bab163801f899";

const client = new ElevenLabsClient({
  apiKey: ELEVENLABS_API_KEY,
});

export const createAudioFileFromText = async (
  text: string
): Promise<string> => {
  return new Promise<string>(async (resolve, reject) => {
    try {
      const audio = await client.voiceGeneration.generate({
        gender: "female",
        accent: "american",
        age: "middle_aged",
        accent_strength: 1.0,
        text,
      });
      const fileName = `${uuid()}.mp3`;
      const buffer: any[] = [];
      const fileStream = new WritableStream({

        write(chunk, controller) {
          buffer.push(chunk);
        },
        close() {
          const bufferData = Buffer.concat(buffer);
          writeFileSync(fileName, bufferData);
          resolve(fileName);
        },
        abort() {
          reject("Stream aborted");
        },
      })
      createWriteStream(fileName);

      // @ts-expect-error
      audio.pipeTo(fileStream);
    } catch (error) {
      reject(error);
    }
  });
};

createAudioFileFromText("Hello, World! Generate some random text that's more than 100 characters long and less than 1000 characters long.");
```

We have one outstanding item to polyfill both `pipe` and `pipeTo` on the returned stream, but works on Node 16 - 21. Please let me know if you have issues, and I'll keep you updated when the full fix comes.

**Comment by dsinghvi** [comment]
@Amerr @lumalisan this should be resolved in `0.13.0`. Please let us know if you still have any issues!
